src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * StreamedJsonResponse represents a streamed HTTP response for JSON.
+ *
+ * A StreamedJsonResponse uses a structure and generics to create an
+ * efficient resource saving JSON response.
+ *
+ * It uses flush after a specified flush size to directly stream the data.
+ *
+ * @see flush()
+ *
+ * @author Alexander Schranz <alexander@sulu.io>
+ *
+ * Example usage:
+ *
+ * function loadArticles(): \Generator
+ *     // some streamed loading
+ *     yield ['title' => 'Article 1'];
+ *     yield ['title' => 'Article 2'];
+ *     yield ['title' => 'Article 3'];
+ * }),
+ *
+ * $response = new StreamedJsonResponse(
+ *     // json structure with generators in which will be streamed
+ *     [
+ *         '_embedded' => [
+ *             'articles' => loadArticles(), // any generator which you want to stream as list of data
+ *         ],
+ *     ],
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    private int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
+
+    /**
+     * @param mixed[]                        $data      JSON Data containing PHP generators which will be streamed as list of data
+     * @param int                            $status    The HTTP status code (200 "OK" by default)
+     * @param array<string, string|string[]> $headers   An array of HTTP headers
+     * @param int                            $flushSize After every which item of a generator the flush function should be called
+     */
+    public function __construct(
+        private readonly array $data,
+        int $status = 200,
+        array $headers = [],
+        private int $flushSize = 100,
+    ) {
+        parent::__construct($this->stream(...), $status, $headers);
+
+        if (!$this->headers->get('Content-Type')) {
+            $this->headers->set('Content-Type', 'application/json');
+        }
+    }
+
+    private function stream(): void
+    {
+        $generators = [];
+        $structure = $this->data;
+
+        array_walk_recursive($structure, function (&$item, $key) use (&$generators) {
+            if ($item instanceof \Generator) {
```

Comment:
you lost support for other types of iterators here

Reply:
what should we do here? we could do if `\Traversable` -> `iterator_to_array`?

Reply:
if `Traversable`, do exactly what you do right now with the generator. The only thing you do with your Generator is using `foreach` on it. And the interface enabling that is `Traversable`, it is not specific to `Generator` (which is a specific implementation of `Iterator` which extends `Traversable`)

Reply:
for traversables s`$generator = fn() => yield from $traversable` would do the trick

Reply:
this would require `(fn () => yield from $traversable)()` to get the generator actually. This is much more complex than supporting Traversable directly (which has strictly no maintenance cost as the existing implementation does not _need_ a generator right now)

Reply:
im opting for closure support out-of-the box yes

Reply:
either way, $generators could be typed `array<string, \Closure|\Traversable>` given generators are traversables

Reply:
Changed the check to \Traversable